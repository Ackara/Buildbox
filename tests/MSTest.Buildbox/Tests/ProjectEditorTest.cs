using Acklann.Buildbox.Versioning;
using Acklann.Buildbox.Versioning.Editors;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.IO;
using System.Linq;

namespace MSTest.Buildbox.Tests
{
    [TestClass]
    public class ProjectEditorTest
    {
        [TestMethod]
        public void Update_should_edit_a_netstardard_project_file_metadata()
        {
            // Arrange
            var manifest = CreateManifest();
            var sut = new NetStandardProjectEditor();

            // Act
            string tempPath = Path.GetTempFileName();
            var projectFile = sut.FindProjectFile(TestDataDirectory.FullPath).First();
            projectFile.CopyTo(tempPath, overwrite: true);
            sut.Update(manifest, new FileInfo(tempPath));

            // Assert
            Approvals.VerifyXml(File.ReadAllText(tempPath));
        }

        [TestMethod]
        public void Update_should_edit_a_dotnet_project_file_metadata()
        {
            // Arrange
            var manifest = CreateManifest();
            var sut = new DotNetProjectEditor();

            // Act
            string tempPath = Path.Combine(Path.GetTempPath(), $"{nameof(Update_should_edit_a_dotnet_project_file_metadata)}.txt");
            var projectFile = sut.FindProjectFile(TestDataDirectory.FullPath).First();
            projectFile.CopyTo(tempPath, overwrite: true);
            sut.Update(manifest, new FileInfo(tempPath));

            // Assert
            Approvals.VerifyFile(tempPath);
        }

        [TestMethod]
        public void Update_should_edit_a_vsix_manifest_file()
        {
            // Arrange
            var manifest = CreateManifest();
            var sut = new VsixManifestEditor();

            // Act
            string tempPath = Path.Combine(Path.GetTempPath(), $"{nameof(Update_should_edit_a_vsix_manifest_file)}.xml");
            var projectFile = sut.FindProjectFile(TestDataDirectory.FullPath).First();
            projectFile.CopyTo(tempPath, overwrite: true);
            sut.Update(manifest, new FileInfo(tempPath));

            // Assert
            Approvals.VerifyFile(tempPath);
        }

        private static Manifest CreateManifest()
        {
            var manifest = new Manifest()
            {
                Title = "Buildbox",
                Version = new VersionInfo(1, 2, 3),
                Authors = "Seto Kiaba",
                Owner = "Kiaba Corp",
                Copyright = "All rights reserved by Me",
                Description = "This is for a automated test",
                ProjectUrl = "https://gihub.com/Ackara/Buildbox",
                IconUri = "https://example.com/icon/buildbox.png",
                LicenseUri = "https://example.com/buildbox/license.txt",
                Tags = "powershell, deploy, ci"
            };

            return manifest;
        }
    }
}