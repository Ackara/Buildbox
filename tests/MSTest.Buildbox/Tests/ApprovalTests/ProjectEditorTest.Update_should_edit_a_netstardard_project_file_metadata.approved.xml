<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard1.4</TargetFramework>
    <AssemblyName>Acklann.Buildbox</AssemblyName>
    <RootNamespace>Acklann.Buildbox</RootNamespace>
    <Product>Buildbox</Product>
    <AssemblyVersion>1.2.3</AssemblyVersion>
    <PackageVersion>1.2.3</PackageVersion>
    <Description>This is for a automated test</Description>
    <Authors>Seto Kiaba</Authors>
    <Company>Kiaba Corp</Company>
    <PackageTags>powershell, deploy, ci</PackageTags>
    <Copyright>All rights reserved by Me</Copyright>
    <PackageIconUrl>https://example.com/icon/buildbox.png</PackageIconUrl>
    <PackageProjectUrl>https://gihub.com/Ackara/Buildbox</PackageProjectUrl>
    <PackageLicenseUrl>https://example.com/buildbox/license.txt</PackageLicenseUrl>
    <PackageReleaseNotes />
  </PropertyGroup>
</Project>